// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: protobuf/api/common.proto

package commerce_api

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on MessageResponse with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *MessageResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on MessageResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// MessageResponseMultiError, or nil if none found.
func (m *MessageResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *MessageResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Message

	if len(errors) > 0 {
		return MessageResponseMultiError(errors)
	}

	return nil
}

// MessageResponseMultiError is an error wrapping multiple validation errors
// returned by MessageResponse.ValidateAll() if the designated constraints
// aren't met.
type MessageResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m MessageResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m MessageResponseMultiError) AllErrors() []error { return m }

// MessageResponseValidationError is the validation error returned by
// MessageResponse.Validate if the designated constraints aren't met.
type MessageResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e MessageResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e MessageResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e MessageResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e MessageResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e MessageResponseValidationError) ErrorName() string { return "MessageResponseValidationError" }

// Error satisfies the builtin error interface
func (e MessageResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sMessageResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = MessageResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = MessageResponseValidationError{}
